add option to lookup multiple coordinates from stdin

DvdGiessen opened the pull request:
The binary is ideal for looking up a timezone in a script. However I needed to look up a lot of them (a few millions), for which the CLI is too slow because it requires starting a new process for each coordinate, which has to initialize the finder from scratch every time.

This MR adds a single extra option to the binary: `--stdin-order {lng-lat,lat-lng}`

It is mutually exclusive with providing a longitude and latitude as options on the command line. Instead, when the binary is started with this option it will read coordinates from the standard input, one coordinate per line, and print a line with the resulting timezone string.

I tried to keep it as contained as possible (only adding a single option, not changing any other source files than for the binary, not requiring any additional dependencies) while making it reasonably fast (reusing a finder, input buffering, avoiding unnecessary allocations).

Open to feedback!

EDIT: Re-pushed because I didn't stage and commit all my changes.

ringsaturn commented:
Hi there,

Thanks for the PR. The tzf-rs' cli is created for debug and haven't releaized that it can used to process millions points.

I have tried this:

```bash
cat coords.txt
```

```console
116.3883 39.9289
114.1594 22.3173
8.61231 47.66148
```

Then run as:

```bash
cat coords.txt| ./target/release/tzf --stdin-order lng-lat
```

```console
"Asia/Shanghai"
"Asia/Hong_Kong"
"Europe/Berlin"
```

The output looks good for me.

Please fix format issue and update README for the cli usege.

DvdGiessen commented:
I've fixed the format issue and added a small description to the README. Thank you for the feedback!

Also added one small change to make it compatible with Windows-style line endings (`\r\n` in addition to only `\n`).

ringsaturn commented:
Release https://github.com/ringsaturn/tzf-rs/releases/tag/v0.4.13
